perllib/FixMyStreet/Roles/Cobrand/Echo.pm

```diff
@@ -32,8 +33,43 @@ requires 'waste_bulky_missed_blocked_codes';
 
 FixMyStreet::Roles::Cobrand::Echo - shared code between cobrands using an Echo backend
 
+=head2 waste_check_downtime
+
+Echo has regular periods of scheduled downtime; we record these
+in a file and check it here to show appropriate messaging.
+
 =cut
 
+sub waste_check_downtime {
+    my $self = shift;
+    my $c = $self->{c};
+
+    my $parser = DateTime::Format::W3CDTF->new;
+    my $now = DateTime->now->set_time_zone(FixMyStreet->local_time_zone);
+
+    my $cfg = $self->feature('echo');
+    my $downtime_csv = $cfg->{downtime_csv};
+    my @lines = eval { path(FixMyStreet->path_to($downtime_csv))->lines({ chomp => 1 }) };
+    foreach (@lines) {
+        my ($start, $end) = map { eval { $parser->parse_datetime($_) } } split /,/;
+        next unless $start && $end;
+        my $start_hour = $start->strftime('%l%P');
+        my $end_hour = $end->strftime('%l%P');
+        my $message = "Due to planned maintenance, our waste management system will be unavailable from $start_hour until $end_hour.";
```

Comment:
I copied it from the down message; yes, it could certainly say "this service" or similar.
It hasn't before; even if it did, it would say "8pm to 1am" which I think is fine.